app-misc/nnn/nnn-4.7.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit bash-completion-r1 flag-o-matic toolchain-funcs xdg-utils
+
+DESCRIPTION="The missing terminal file browser for X"
+HOMEPAGE="https://github.com/jarun/nnn"
+SRC_URI="https://github.com/jarun/nnn/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE="+readline pcre 8contexts icons nerdfonts emoji qsort"
+
+DEPEND="sys-libs/ncurses:=
+	pcre? ( dev-libs/libpcre )
+	readline? ( sys-libs/readline:= )
+	elibc_musl? ( sys-libs/fts-standalone )"
+RDEPEND="${DEPEND}"
+REQUIRED_USE="?? ( icons nerdfonts emoji )"
```

Comment:
I'm not a huge fan of what's happening in here. Always try to avoid `REQUIRED_USE` whenever possible. In this case, users have to manually do steps outside setting use flags to get these to work anyway.

So my question is: does the build system prevent from these being toggled on? Since in this ebuild they don't pull any extra deps, can they be enabled conditionally and then put a `pkg_postinst` message saying enable one by symlinking it or something?

Reply:
It's a build time toggle, yes. And the options are mutually exclusive with each other.

Reply:
The build system will not allow enabling more than one of these. As far as I know, the respective icons are enabled unconditionally when nnn is built with support for them. They may not be displayed *correctly* (depending on the font setup), but *something* will show. Therefore, we need to option disable any single or none these flags.
If there is anything better suited than `REQUIRED_USE`, I'll happily switch to that. At the moment, I can't think of an alternative.

Reply:
You can "prioritize" the choice based on use flags selection - most likely just how upstream does it. But in this case since there's the additional `pkg_postinst` steps users have to do, I believe it'd be a messier user experience.